### PR TITLE
Add gelsons spider

### LINKED ISSUE #3532: Gelson's
https://www.gelsons.com/stores

### PR DESCRIPTION
**Pull request by cmecklenborg**
Fixes #3532

### CODE REVIEW

File: locations/spiders/gelsons.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+                )
+
+    def parse_store(self, response):
+
+        content = json.loads(
+            response.xpath('//script[@type="application/json"]/text()').extract_first()
+        )["props"]["pageProps"]
+        store_json = content["store"]
+        hours = content["pageComponents"][0]["headline"]
+
```
Comment (mjoe999) [inline locations/spiders/gelsons.py:33]: Hi,
item=DictParser.parse(store_json) should do quite a bit of what you have typed in below. Bits it does not do then you can still do! Bit like this:
https://github.com/alltheplaces/alltheplaces/blob/master/locations/spiders/greggs_gb.py
Gives your keyboard a bit more life :-) Thank you for the PR.
Cheers,
Joe.

File: locations/spiders/gelsons.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    def parse(self, response):
+        response.selector.remove_namespaces()
+        urls = response.xpath("//url/loc/text()").extract()
+        for url in urls:
+            if re.match(r".*stores/.+$", url) and "virtual" not in url:
+
+                yield scrapy.Request(
+                    url=url, callback=self.parse_store, meta={"url": url}
+                )
```
Comment (Cj-Malone) [inline locations/spiders/gelsons.py:24]: This should be a SitemapSpider, you may need to use `sitemap_filter` to remove the virtual store. See [HalifaxGB](https://github.com/alltheplaces/alltheplaces/blob/0562b1ccd19d41f9fa835a20bd262a2173ecb2e8/locations/spiders/halifax_gb.py) as an example.

File: locations/spiders/gelsons.py (hunk 1 above)
Comment (cmecklenborg) [reply locations/spiders/gelsons.py:33]: Thanks for the tip!

File: locations/spiders/gelsons.py (hunk 2 above)
Comment (cmecklenborg) [reply locations/spiders/gelsons.py]: Great, thanks! Not sure how I missed this class, given how many sitemap-based spiders I've written. 😅

**Review by Cj-Malone — APPROVED** [review]

**Pull request merged**